src/dotnet-publish-iis/WebConfigTransform.cs

```diff
@@ -64,6 +64,25 @@ private static void TransformHttpPlatform(XElement httpPlatformElement, string a
             SetAttributeValueIfEmpty(httpPlatformElement, "stdoutLogEnabled", "false");
             SetAttributeValueIfEmpty(httpPlatformElement, "stdoutLogFile", logPath);
             SetAttributeValueIfEmpty(httpPlatformElement, "startupTimeLimit", "3600");
+
+            AddApplicationBase(httpPlatformElement);
+        }
+
+        private static void AddApplicationBase(XElement httpPlatformElement)
+        {
+            const string appBaseKeyName = "ASPNET_APPLICATIONBASE";
+
+            var envVariables = GetOrCreateChild(httpPlatformElement, "environmentVariables");
+            var appBaseElement = envVariables.Elements("environmentVariable").SingleOrDefault(e =>
+                string.Equals((string)e.Attribute("name"), appBaseKeyName, StringComparison.CurrentCultureIgnoreCase));
+
+            if (appBaseElement == null)
```

Comment:
I meant test that verifies case where web.config you are modifying already has ASPNET_APPLICATIONBASE and you shouldn't not change it.